lib/src/zenohRpcClient.cpp

```diff
@@ -112,121 +117,215 @@ UStatus ZenohRpcClient::term() noexcept {
 
     return status;
 }
-std::future<UPayload> ZenohRpcClient::invokeMethod(const UUri &uri, 
-                                                   const UPayload &payload, 
-                                                   const UAttributes &attributes) noexcept {
-    std::future<UPayload> future;
+
+std::future<RpcResponse> ZenohRpcClient::invokeMethod(const UUri &topic, 
+                                                      const UPayload &payload, 
+                                                      const CallOptions &options) noexcept {
+    std::future<RpcResponse> future;
 
     if (0 == refCount_) {
         spdlog::error("ZenohRpcClient is not initialized");
         return future;
     }
 
-    if (false == isRPCMethod(uri.resource())) {
+    if (false == isRPCMethod(topic.resource())) {
         spdlog::error("URI is not of RPC type");
```

Comment:
Maybe more a general question on philosophy -- when a function such as `invokeMethod()` or `send()` doesn't complete successfully, does it make sense to more explicitly model that interaction as returning an exceptional result?

For example here, the failure is logged, but is there any expectation that a developer writing a uE should be able to "understand" when these functions fail and why and have that modeled in the code so they can respond to the failure?

Or is there instead going to be documentation on what shall be returned when a failure occurred that developers read and understand?

Reply:
I think that we need to add more documentation

Reply:
What do you expect to return then ? invalid uSTATUS?